benches/bench_basic.rs

```diff
@@ -35,21 +33,23 @@ fn bench_simple_getsetdel_async(b: &mut Bencher) {
     let client = get_client();
     let mut runtime = Runtime::new().unwrap();
     let con = client.get_async_connection();
-    let mut opt_con = Some(runtime.block_on(con).unwrap());
+    let mut con = runtime.block_on(con).unwrap();
 
     b.iter(|| {
-        let con = opt_con.take().expect("No connection");
-
-        let key = "test_key";
-        let future = redis::cmd("SET")
-            .arg(key)
-            .arg(42)
-            .query_async(con)
-            .and_then(|(con, ())| redis::cmd("GET").arg(key).query_async(con))
-            .and_then(|(con, _): (_, isize)| redis::cmd("DEL").arg(key).query_async(con));
-        let (con, ()) = runtime.block_on(future).unwrap();
-
-        opt_con = Some(con);
+        runtime
+            .block_on(async {
+                let key = "test_key";
+                let () = redis::cmd("SET")
+                    .arg(key)
+                    .arg(42)
+                    .query_async(&mut con)
+                    .await?;
+                let _: isize = redis::cmd("GET").arg(key).query_async(&mut con).await?;
+                let () = redis::cmd("DEL").arg(key).query_async(&mut con).await?;
+                Ok(())
+            })
+            .map_err(|err: RedisError| err)
```

Comment:
I'm not sure what this `map_err()` is for?

Reply:
It lets rustc know which error type to use (due to `From` use in `?`).